net/znc/files/znc.init

```diff
@@ -156,45 +155,27 @@ add_user() {
 	echo "</User>" >> $ZNC_CONFIG
 }
 
-
-start() {
+start_service() {
 	config_load znc
 	config_foreach znc_global znc
 
-	if [ "$DISABLED" -eq 1 ]; then
-		return 0
-	fi
+	[ "$DISABLED" -eq 0 ] || return 0
 
 	if [ "$EXTERNAL_CONFIG" -eq 0 ]
 	then
 		config_foreach add_listener listener
 		config_foreach add_user user
 
-		chown -hR ${RUNAS_USER:-nobody}:${RUNAS_GROUP:-nogroup} ${ZNC_CONFIG_PATH}
 	fi
 
-	if [ "$EXTERNAL_CONFIG" -eq 1 -a "$RUNAS_USER"  ]
-	then
-		local SU=$(which su)
-		if [ "$SU" ]
-		then
-			chown -hR ${RUNAS_USER:-nobody}:${RUNAS_GROUP:-nogroup} ${ZNC_CONFIG_PATH}
-			$SU ${RUNAS_SHELL:+s $RUNAS_SHELL} -c "/usr/bin/znc -d$ZNC_CONFIG_PATH >/dev/null &" $RUNAS_USER
-		else
-			logger -s -t ZNC -p daemon.err "Could not run ZNC as user $RUNAS_USER: su not found."
-			exit 1
-		fi
-	else
-		/usr/bin/znc -d$ZNC_CONFIG_PATH >/dev/null &
-	fi
-}
+	chown -hR ${RUNAS_USER}:${RUNAS_GROUP} ${ZNC_CONFIG_PATH}
 
-stop() {
-	if [ -f "$PID_FILE" ]
-	then
-		kill $(cat "$PID_FILE")
-	else
-		killall znc
-	fi
+	procd_open_instance
+	procd_set_param file /etc/config/znc
+	[ "$EXTERNAL_CONFIG" -eq 1 ] && procd_set_param file "${ZNC_CONFIG}/configs/znc.conf"
+	procd_set_param command /usr/bin/znc
+	procd_append_param command -f -d$ZNC_CONFIG_PATH
+	procd_set_param user ${RUNAS_USER:-nobody}
```

Comment:
Thinking about it, probably not.  It was an ill thought out idea to cope with a missing user ID.  I've removed it.